cluster/swarm/cluster.go

```diff
@@ -196,10 +199,15 @@ func (c *Cluster) RemoveImage(image *cluster.Image) ([]*dockerclient.ImageDelete
 
 // Pull is exported
 func (c *Cluster) Pull(name string, callback func(what, status string)) {
-	size := len(c.engines)
-	done := make(chan bool, size)
+	var wg sync.WaitGroup
+
+	c.RLock()
 	for _, n := range c.engines {
+		wg.Add(1)
```

Comment:
With wg.Add in the loop the wg is a part of the task distribution. 
If we decided to launch less goroutines then length of c.engines (e.g. via condition), it would lead to a deadlock.
Also it's better to remove a blank line between wg.Add & go func.

The last argument: http://golang.org/pkg/sync/#example_WaitGroup :D